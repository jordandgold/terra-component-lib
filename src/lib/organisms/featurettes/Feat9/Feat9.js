import React from "react";
import "./Feat9.scss";
import PropTypes from "prop-types";

const Feat9 = props => {
  const { image, quote } = props;
  return (
    <section className="ter-feat9">
      <img className="ter-feat9--image" src={image.url} alt={image.altText} />
      <img
        className="ter-feat9--quote-icon"
        src="./quote.png"
        alt="downward quote"
      />
      <div className="ter-feat9--content">
        <p className="ter-feat9--quote-text">{quote.text}</p>
        <p className="ter-feat9--quote-source">- {quote.source}</p>
        <p className="ter-feat9--quote-source-title">{quote.sourceTitle}</p>
      </div>
    </section>
  );
};

export default Feat9;

Feat9.propTypes = {
  imageSide: PropTypes.string,
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
  }).isRequired,
  quote: PropTypes.shape({
    text: PropTypes.string.isRequired,
    source: PropTypes.string,
    sourceTitle: PropTypes.string
  }).isRequired
};
