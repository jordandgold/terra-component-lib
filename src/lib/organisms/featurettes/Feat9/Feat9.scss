@import "../../../global-styles/terra.scss";

.ter-feat9 {
  margin: spacing("none") spacing("auto");
  padding: spacing("none");
  max-width: 1400px;
  display: grid;
  grid-template-columns: 350px 1fr;

  &__image {
    width: 300px;
    height: 300px;
    padding-top: spacing("md");
    box-sizing: content-box;
    justify-self: end;
    display: block;
  }

  &__full-quote-wrapper {
    display: flex;
    margin-left: spacing("lg");
  }

  &__quote-icon {
    height: spacing("md");
    margin-top: spacing("md");
  }

  &__content {
    margin: spacing("md") spacing("xs");
  }

  &__quote-source {
    font-weight: font-weight("bold");
    margin-bottom: spacing("xs");
    margin-left: spacing("mini");
  }
  &__quote-source-title {
    color: color("gray-6");
    font-size: font-size("micro");
    text-transform: uppercase;
    margin-left: spacing("mini");
  }
}

@media (max-width: 900px) {
  .ter-feat9 {
    margin: spacing("none") spacing("auto");
    padding: spacing("none");
    max-width: 400px;
    display: block;
    grid-template-columns: 1fr;

    &__image {
      // margin: spacing('auto');
      width: 200px;
      height: 200px;
      margin-left: spacing("lg");
    }

    &__full-quote-wrapper {
      max-width: 400px;
      margin: spacing("auto");
    }
  }
}

@media (max-width: 600px) {
  .ter-feat9 {
    &__full-quote-wrapper {
      max-width: 300px;
    }

    &__image {
      margin: spacing("auto");
    }
  }
}
