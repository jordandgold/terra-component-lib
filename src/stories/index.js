import React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { linkTo } from "@storybook/addon-links";
import Button from "../lib/Button/Button";
import Alert from "../lib/Alert/Alert";
import Modal from "../lib/Modal/Modal";
import Notification from "../lib/Notification/Notification";
import Dropdown from "../lib/Dropdown/Dropdown";
import Icon from "../lib/Icon/Icon";
import Pagination from "../lib/Pagination/Pagination";
import Breadcrumbs from "../lib/Breadcrumbs/Breadcrumbs";
import TextInput from "../lib/TextInput/TextInput";
import Radios from "../lib/Radios/Radios";
import TextArea from "../lib/TextArea/TextArea";
import Checkbox from "../lib/Checkbox/Checkbox";
import Select from "../lib/Select/Select";
import SearchSelect from "../lib/SearchSelect/SearchSelect";
import MultipleSearchSelect from "../lib/MultipleSearchSelect/MultipleSearchSelect";
import Table from "../lib/Table/Table";

// import { Button, Welcome } from "@storybook/react/demo";

// storiesOf("Welcome", module).add("to Storybook", () => (
//   <Welcome showApp={linkTo("Button")} />
// ));

// storiesOf("Button", module)
//   .add("with text", () => (
//     <Button onClick={action("clicked")}>Hello Button</Button>
//   ))
//   .add("with some emoji", () => (
//     <Button onClick={action("clicked")}>
//       <span role="img" aria-label="so cool">
//         😀 😎 👍 💯
//       </span>
//     </Button>
//   ));

storiesOf("Button", module)
  .add("Primary Buttons", () => (
    <div>
      <Button
        onClick={action("Primary 1 Clicked")}
        text="Primary 1"
        className="ter-button--primary--1"
      />
      <Button
        onClick={action("Primary 2 Clicked")}
        text="Primary 2"
        className="ter-button--primary--2"
      />
      <Button
        onClick={action("Primary 3 Clicked")}
        text="Primary 3"
        className="ter-button--primary--3"
      />
      <Button
        onClick={action("Primary 4 Clicked")}
        text="Primary 4"
        className="ter-button--primary--4"
      />
      <Button
        onClick={action("Primary 5 Clicked")}
        text="Primary 5"
        className="ter-button--primary--5"
      />
    </div>
  ))
  .add("Secondary Buttons", () => (
    <div>
      <Button
        onClick={action("Secondary 1 Clicked")}
        text="Secondary 1"
        className="ter-button--secondary--1"
      />
      <Button
        onClick={action("Secondary 2 Clicked")}
        text="Secondary 2"
        className="ter-button--secondary--2"
      />
      <Button
        onClick={action("Secondary 3 Clicked")}
        text="Secondary 3"
        className="ter-button--secondary--3"
      />
      <Button
        onClick={action("Secondary 4 Clicked")}
        text="Secondary 4"
        className="ter-button--secondary--4"
      />
      <Button
        onClick={action("Secondary 5 Clicked")}
        text="Secondary 5"
        className="ter-button--secondary--5"
      />
    </div>
  ));

storiesOf("Alert", module).add("Alerts", () => (
  <div>
    <Alert
      onClick={action("Clicked!")}
      text="Default Alert"
      closeAlert={action("close!")}
    />
    <Alert
      onClick={action("Clicked!")}
      text="Warning Alert"
      type="warning"
      closeAlert={action("close!")}
    />
    <Alert
      onClick={action("Clicked!")}
      text="Danger Alert"
      type="danger"
      closeAlert={action("close!")}
    />
    <Alert
      onClick={action("Clicked!")}
      text="Success Alert"
      type="success"
      closeAlert={action("close!")}
    />
  </div>
));

const mockBody =
  "Must go faster... go, go, go, go, go! Jaguar shark! So tell me - does it really exist? Just my luck, no ice. What do they got in there? King Kong? Yeah, but John, if The Pirates of the Caribbean breaks down, the pirates don’t eat the tourists. So you two dig up, dig up dinosaurs?";

storiesOf("Modal", module).add("Modal", () => (
  <Modal
    closeModal={action("close!")}
    title="Modal title"
    body={mockBody}
    buttonOne={{
      onClick: action("Clicked!"),
      className: "ter-button--primary--1",
      text: "buy sketchup!"
    }}
    buttonTwo={{
      onClick: action("Clicked!"),
      className: "ter-button--secondary--1",
      text: "buy sketchup, dummy!"
    }}
  />
));

storiesOf("Notifcation", module).add("Notification", () => (
  <Notification onClick={action("clicked!")} type="default" text={mockBody} />
));

storiesOf("Icon", module).add("Icon", () => (
  <div>
    <Icon type="open-caret-right-dark-8px" size="8px" />
    <Icon type="open-caret-up-dark-8px" size="8px" />
    <Icon type="open-caret-down-dark-8px" size="8px" />
    <Icon type="open-caret-left-dark-8px" size="8px" />
  </div>
));

const mockDropdownOptions = ["Option 1", "Option 2", "Option 3"];

storiesOf("Dropdown", module).add("Dropdown", () => (
  <Dropdown
    defaultLabel="Select an option"
    options={mockDropdownOptions}
    selectOption={action("Option selected!")}
  />
));

const mockPages = 4;

storiesOf("Pagination", module).add("Pagination", () => (
  <Pagination activePage={0} pages={mockPages} />
));

const mockBreadcrumbs = [
  { link: "test", text: "Breadcrumb" },
  { link: "test", text: "Breadcrumb" },
  { link: "test", text: "Breadcrumb" }
];

storiesOf("Breadcrumb", module).add("Breadcrumb", () => (
  <Breadcrumbs breadcrumbs={mockBreadcrumbs} />
));

storiesOf("TextInput", module).add("TextInput", () => (
  <div>
    <TextInput
      placeholder="placeholder text"
      value=""
      inputChange={action("input received")}
      name="test"
    />
  </div>
));

storiesOf("Radios", module).add("Radios", () => {
  const mockRadios = [
    { name: "label one" },
    { name: "label two" },
    { name: "label three" }
  ];
  return (
    <div>
      <Radios
        radios={mockRadios}
        selected={"label one"}
        selectRadio={action("radio clicked")}
        collection="storybook-radios"
      />
    </div>
  );
});

storiesOf("TextArea", module).add("TextArea", () => {
  return (
    <TextArea handleChange={action("ohh, text!")} value="" label="Text Area" />
  );
});

storiesOf("Checkbox", module).add("Checkbox", () => {
  return (
    <Checkbox
      name="checkbox"
      label="This is a checkbox"
      checked={false}
      handleChange={action("checked/unchecked")}
    />
  );
});

storiesOf("SearchSelect", module).add("SearchSelect", () => {
  const mockOptions = [
    "cats",
    "dogs",
    "turtles",
    "fish",
    "ferrets",
    "hamsters",
    "birds"
  ];

  return (
    <SearchSelect
      defaultText="This is a SearchSelect component"
      options={mockOptions}
      selection={undefined}
      handleSelect={action("selection made")}
    />
  );
});

storiesOf("MultipleSearchSelect", module).add("MultipleSearchSelect", () => {
  const mockOptions = [
    "cats",
    "dogs",
    "turtles",
    "fish",
    "ferrets",
    "hamsters",
    "birds"
  ];

  return (
    <MultipleSearchSelect
      defaultText="This is a MultipleSearchSelect component"
      options={mockOptions}
      selections={[]}
      handleSelect={action("selection made")}
      removeSelection={action("remove selection")}
    />
  );
});

storiesOf("Select", module).add("Select", () => {
  const mockOptions = [
    "cats",
    "dogs",
    "turtles",
    "fish",
    "ferrets",
    "hamsters",
    "birds"
  ];

  return (
    <Select
      options={mockOptions}
      handleSelection={action("clicked an option")}
      selection={undefined}
      defaultText="This is a Select Component"
    />
  );
});

storiesOf("Table", module).add("Table", () => {
  const mockData = {
    head: ["Table Head 1", "Table Head 2", "Table Head 3"],
    body: [
      [["Row 1 Cell 1"], ["Row 1 Cell 2"], ["Row 1 Cell 3"]],
      [["Row 2 Cell 1"], ["Row 2 Cell 2"], ["Row 2 Cell 3"]],
      [["Row 3 Cell 1"], ["Row 3 Cell 2"], ["Row 3 Cell 3"]]
    ]
  };

  return <Table data={mockData} />;
});
