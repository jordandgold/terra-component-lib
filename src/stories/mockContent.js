export const heroTwoContent = {
  images: {
    imageThreeXTwo: {
      url: "./3x2-ph.png",
      altText: "placeholder"
    },
    imageFourXThree: {
      url: "./800x600.png",
      altText: "placeholder"
    },
    imageOneXOne: {
      url: "./1-to-1.png",
      altText: "placeholder"
    }
  },
  title: "Good design is innovative.",
  text:
    "Good design is aesthetic. Good design makes a product understandable. Good design is unobtrusive.",
  ctas: {
    ctaOne: {
      className: "ter-button--primary--1",
      link: "#",
      text: "Button"
    },
    ctaTwo: {
      className: "ter-button--secondary--1",
      link: "#",
      text: "Button"
    },
    subCTA: {
      text: "Sub-CTA",
      link: "#"
    }
  }
};

export const featSixAContent = {
  title: "Good design is innovative.",
  ctas: {
    ctaOne: {
      className: "ter-button--primary--1",
      name: "ctaOne",
      text: "Button"
    },
    ctaTwo: {
      className: "ter-button--secondary--1",
      name: "ctaOne",
      text: "Button"
    }
  }
};

export const featNineContent = {
  image: {
    url: "./1-to-1.png",
    altText: "dock on water"
  },
  imageSide: "right",
  quote: {
    text:
      "Using SketchUp Studio's Sefiara Software, we were able to study multiple options within a short period of time, which allowed the team to arrive at a design direction that we all felt comfortable with much faster with meaningful data as opposed to 'rules of thumb' and broad assumptions.",
    source: "Leonard Sciarra",
    sourceTitle: "Senior Associate at Gensler, AIA, ASHRAE, LEED AP+"
  }
};

export const featSevenContent = {
  title: "Design is cool, I guess",
  content: [
    {
      title: "Design!",
      body:
        "Enim laborum ad anim laborum ad minim ipsum proident est cillum aliqua. Irure laboris aute ullamco ad Lorem et culpa id commodo quis sunt labore in id.",
      link: "#"
    },
    {
      title: "Aesthetics!",
      body:
        "Enim laborum ad anim laborum ad minim ipsum proident est cillum aliqua.",
      link: "#"
    },
    {
      title: "Useability!",
      body:
        "Enim laborum ad anim laborum ad minim ipsum proident est cillum aliqua. Qui proident ipsum excepteur aute eiusmod excepteur commodo pariatur mollit officia. Dolore occaecat tempor do in officia elit deserunt cillum sunt labore ad tempor.",
      link: "#"
    },
    {
      title: "Coffee!",
      body:
        "Enim laborum ad anim laborum ad minim ipsum proident est cillum aliqua.",
      link: "#"
    }
  ]
};

export const featFourContent = [
  {
    title: "Design is good. Period.",
    body:
      "Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!",
    link: {
      url: "#",
      text: "Button"
    }
  },
  {
    title: "Design is good. Period.",
    body:
      "Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!",
    link: {
      url: "#",
      text: "Button"
    }
  },
  {
    title: "Design is good. Period.",
    body:
      "Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!",
    link: {
      url: "#",
      text: "Button"
    }
  },
  {
    title: "Design is good. Period.",
    body:
      "Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!",
    link: {
      url: "#",
      text: "Button"
    }
  }
];

export const mockDumplings = [
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  }
];

export const mockImageDumplings = [
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  }
];

export const mockTestIconDumplings = [
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    size: "small",
    icon: { type: "enclosed-check-dark-48px", size: "48px" }
  }
];

export const mockTestImageDumplings = [
  {
    title: "Dumplin'",
    link: { url: "#" },
    dumplingSize: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  },
  {
    title: "Dumplin'",
    link: { url: "#" },
    dumplingSize: "small",
    image: { url: "./1-to-1.png", altText: "test text" }
  }
];

export const mockButtonCardContent = {
  title: "This is a card title",
  text: "Esse adipisicing commodo cupidatat fugiat.",
  image: { url: "./placeholder.jpg", altText: "placeholder image" },
  button: { text: "Do Something", name: "name" },
  size: "small"
};

export const mockButtonLinkCardContent = {
  title: "This is a card title",
  text: "Esse adipisicing commodo cupidatat fugiat.",
  image: { url: "./placeholder.jpg", altText: "placeholder image" },
  buttonLink: { text: "Go Somewhere", link: "#" },
  size: "small"
};

export const mockLinkCardContent = {
  title: "This is a card title",
  text: "Esse adipisicing commodo cupidatat fugiat.",
  image: { url: "./placeholder.jpg", altText: "placeholder image" },
  link: { url: "#", text: "Go Somewhere" },
  size: "small"
};

export const mockCards = [
  {
    image: {
      url: "./placeholder.jpg",
      altText: "Placeholder"
    },
    title: "Card Title",
    text:
      "Irure cillum ad culpa ad non cillum irure dolore ad nostrud aliqua mollit. Eiusmod id mollit non voluptate qui qui ut ullamco officia eiusmod laboris veniam amet culpa."
  },
  {
    image: {
      url: "./placeholder.jpg",
      altText: "Placeholder"
    },
    title: "Card Title",
    text:
      "Irure cillum ad culpa ad non cillum irure dolore ad nostrud aliqua mollit. Eiusmod id mollit non voluptate qui qui ut ullamco officia eiusmod laboris veniam amet culpa."
  },
  {
    image: {
      url: "./placeholder.jpg",
      altText: "Placeholder"
    },
    title: "Card Title",
    text:
      "Irure cillum ad culpa ad non cillum irure dolore ad nostrud aliqua mollit. Eiusmod id mollit non voluptate qui qui ut ullamco officia eiusmod laboris veniam amet culpa."
  }
];
