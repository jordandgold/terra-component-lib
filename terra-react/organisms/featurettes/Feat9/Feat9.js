import React from "react";
import "./Feat9.scss";

var Feat9 = function Feat9(props) {
  var image = props.image,
      quote = props.quote;
  return React.createElement("section", {
    className: "ter-feat9"
  }, React.createElement("img", {
    className: "ter-feat9--image",
    src: image.url,
    alt: image.altText
  }), React.createElement("div", {
    className: "ter-feat9--full-quote-wrapper"
  }, React.createElement("img", {
    className: "ter-feat9--quote-icon",
    src: "./quote.png",
    alt: "downward quote"
  }), React.createElement("div", {
    className: "ter-feat9--content"
  }, React.createElement("p", {
    className: "ter-feat9--quote-text"
  }, quote.text), React.createElement("p", {
    className: "ter-feat9--quote-source"
  }, "- ", quote.source), React.createElement("p", {
    className: "ter-feat9--quote-source-title"
  }, quote.sourceTitle))));
};

export default Feat9;