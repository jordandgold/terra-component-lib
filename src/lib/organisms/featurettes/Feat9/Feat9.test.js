import React from "react";
import Feat9 from "./Feat9";
import { shallow } from "enzyme";
import { featNineContent } from "../../../../stories/mockContent";

describe("Feat9", () => {
  let wrapper;
  const { image, imageSide, quote } = featNineContent;

  wrapper = shallow(
    <Feat9 image={image} imageSide={imageSide} quote={quote} />
  );

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
