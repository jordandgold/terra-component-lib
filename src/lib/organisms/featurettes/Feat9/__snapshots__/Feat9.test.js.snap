// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feat9 should match the snapshot 1`] = `
<section
  className="ter-feat9"
>
  <img
    alt="dock on water"
    className="ter-feat9--image"
    src="./1-to-1.png"
  />
  <div
    className="ter-feat9--full-quote-wrapper"
  >
    <img
      alt="downward quote"
      className="ter-feat9--quote-icon"
      src="./ico-quote.svg"
    />
    <div
      className="ter-feat9--content"
    >
      <p
        className="ter-feat9--quote-text"
      >
        Using SketchUp Studio's Sefiara Software, we were able to study multiple options within a short period of time, which allowed the team to arrive at a design direction that we all felt comfortable with much faster with meaningful data as opposed to 'rules of thumb' and broad assumptions.
      </p>
      <p
        className="ter-feat9--quote-source"
      >
        - 
        Leonard Sciarra
      </p>
      <p
        className="ter-feat9--quote-source-title"
      >
        Senior Associate at Gensler, AIA, ASHRAE, LEED AP+
      </p>
    </div>
  </div>
</section>
`;
